frontend/src/tests/ddah-tests.js

```diff
@@ -406,13 +513,80 @@ export function ddahsEmailAndDownloadTests(api) {
     });
 
     it("can download pdf versions ddah signature list", async () => {
-        let resp = await apiGET(
+        const resp = await apiGET(
             `/admin/sessions/${session.id}/ddahs/accepted_list.pdf`
         );
         expect(resp).toHaveStatus("success");
     });
 
     // For this test, it is sufficient to only check the HTML version,
     // which can be downloaded by leaving off the `.pdf` suffix from the public route.
-    it.todo("Downloaded ddah includes all duties (from all categories)");
+    it("Downloaded ddah includes all duties (from all categories)", async () => {
+        const ddahWithAllDutyTypes = {
+            // Sometimes the order of instructors displayed is different
+            // Assignment #4 only has one instructor, so we choose that one
+            // to focus on testing what really matters - duties rendered
+            assignment_id: assignments[3].id,
+            duties: [
+                {
+                    order: 2,
+                    hours: 25,
+                    description: "marking:Marking the test",
+                },
+                {
+                    order: 1,
+                    hours: 4,
+                    description: "training:TA training",
+                },
+                {
+                    order: 3,
+                    hours: 40,
+                    description: "contact:Tutorials",
+                },
+                {
+                    order: 5,
+                    hours: 18,
+                    description: "note:Some note",
+                },
+                {
+                    order: 6,
+                    hours: 4,
+                    description: "meeting:Weekly meetings",
+                },
+                {
+                    order: 7,
+                    hours: 4,
+                    description: "prep:Assignment prep",
+                },
+                {
+                    order: 4,
+                    hours: 10,
+                    description: "other:Other responsibilities",
+                },
+            ],
+        };
+
+        const resp = await apiPOST(`/admin/ddahs`, ddahWithAllDutyTypes);
+        expect(resp).toHaveStatus("success");
+
+        const ddahHtml = (
+            await axios.get(
+                `${BACKEND_BASE_URL}/public/ddahs/${resp.payload.url_token}`
+            )
+        ).data;
+
+        const expectedStrings = [
+            "Training",
+            "Marking/Grading",
+            "Contact Time",
+            "Preparation",
+            "Meetings",
+            "Notes",
+            "Other duties",
+        ];
+
+        for (const expectedString of expectedStrings) {
+            expect(ddahHtml).toEqual(expect.stringContaining(expectedString));
```

Comment:
The labels are not the important part. It's the duty descriptions which are important. Currently, if you make `duties` an empty array, your test will still pass.

Reply:
:open_mouth:  Thanks for pointing that out!